src/pages/FarmV3/FarmTableComponents/FarmItems.tsx

```diff
@@ -0,0 +1,53 @@
+import { SSLToken } from '@/pages/FarmV3/constants'
+import React, { FC, useState } from 'react'
+import { Accordion, AccordionItem } from 'gfx-component-lib'
+import { StatsModal } from '@/pages/FarmV3/StatsModal'
+import NoResultsFound from '@/pages/FarmV3/FarmTableComponents/NoResultsFound'
+import FarmContent from '@/pages/FarmV3/FarmTableComponents/FarmTableBalanceItem'
+import { useSSLContext } from '@/context'
+import { truncateBigString } from '@/utils'
+
+const FarmItems: FC<{
+  tokens: SSLToken[]
+  numberOfCoinsDeposited: number
+  searchTokens: string
+  showDeposited: boolean
+}> = ({ tokens, numberOfCoinsDeposited, searchTokens, showDeposited }) => {
+  const [statsModal, setStatsModal] = useState<boolean>(false)
+  const { filteredLiquidityAccounts } = useSSLContext()
+  return (
+    <div className={''}>
+      {numberOfCoinsDeposited === 0 && showDeposited && searchTokens?.length === 0 && (
+        <NoResultsFound
+          str="Oops, no pools deposited"
+          subText="Don’t worry, explore our pools and start earning!"
+        />
+      )}
+      {tokens?.length === 0 && searchTokens?.length > 0 && (
+        <NoResultsFound
+          requestPool={true}
+          str="Oops, no pools found"
+          subText="Don’t worry, there are more pools coming soon..."
+        />
+      )}
+      <Accordion type={'multiple'} collapsible={true} variant={'secondary'} className={'lg:min-w-full gap-3.75'}>
+        {tokens.map((coin) => {
+          const userDepositedAmount = truncateBigString(
+            filteredLiquidityAccounts?.[coin?.mint?.toBase58()]?.amountDeposited?.toString(),
```

Comment:
on line 43, 37, 36, 45 you do coin chain check. In Typescript it transpiles to huge boilerplate

Reply:
good point early termination implemented